### PR TITLE
Updated packages

### PR DESCRIPTION
**Pull request by MagnetonIO**
Updated Django to 3.1.1 and all other packages to the latest stable release.

### CODE REVIEW

File: apps/jurisdiction/migrations/0001_initial.py [hunk 1]
```diff
@@ -54,6 +54,6 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='jurisdiction',
             name='state',
-            field=models.ForeignKey(to='jurisdiction.State'),
+            field=models.ForeignKey(to='jurisdiction.State', on_delete=models.CASCADE),
```
Comment (crwallace) [inline apps/jurisdiction/migrations/0001_initial.py:57]: Can you tell me a little bi about why use `CASCADE` here. Makes me a little nervous.

File: apps/survey/views.py [hunk 2]
```diff
@@ -2,7 +2,8 @@
 from rest_framework import viewsets
 from rest_framework.response import Response
 from rest_framework.permissions import AllowAny
-from rest_framework.decorators import list_route
+# from rest_framework.decorators import list_route
```
Comment (crwallace) [inline apps/survey/views.py]: Can we remove this comment?

File: apps/survey/views.py [hunk 3]
```diff
@@ -30,7 +31,8 @@ class ContactViewSet(viewsets.ViewSet):
         5: '65 and older'
     }
 
-    @list_route(methods=['post'])
+    # @list_route(methods=['post'])
```
Comment (crwallace) [inline apps/survey/views.py]: Same here

File: apps/survey/views.py [hunk 4]
```diff
@@ -54,7 +56,8 @@ def us(self, request):
         else:
             return Response({'detail': 'Nothing was sent'}, status=status.HTTP_400_BAD_REQUEST)
 
-    @list_route(methods=['post'])
+    # @list_route(methods=['post'])
```
Comment (crwallace) [inline apps/survey/views.py]: Same here

File: apps/survey/views.py [hunk 5]
```diff
@@ -98,7 +101,8 @@ def survey(self, request):
 
         return Response({'detail': 'Thank you.'}, status=status.HTTP_200_OK)
 
-    @list_route(methods=['post'])
+    # @list_route(methods=['post'])
```
Comment (crwallace) [inline apps/survey/views.py]: Same here

File: apps/survey/views.py [hunk 6]
```diff
@@ -183,15 +187,17 @@ def application(self, request):
 
         return Response({'detail': 'Thank you.'}, status=status.HTTP_200_OK)
 
-    @list_route()
+    # @list_route()
```
Comment (crwallace) [inline apps/survey/views.py]: Same here

File: apps/survey/views.py [hunk 7]
```diff
@@ -183,15 +187,17 @@ def application(self, request):
...
+    @action(detail=False)
     def applications_export(self, request):
         if request.user.is_authenticated():
             return export_applications()
         else:
             return Response({'detail': 'Not allowed'},
                             status=status.HTTP_401_UNAUTHORIZED)
 
-    @list_route()
+    # @list_route()
```
Comment (crwallace) [inline apps/survey/views.py]: Same here

File: config/settings/common.py [hunk 8]
```diff
@@ -55,17 +55,30 @@
...
+#     'django.middleware.common.CommonMiddleware',
+#     'django.middleware.csrf.CsrfViewMiddleware',
+#     'django.contrib.auth.middleware.AuthenticationMiddleware',
+#     'django.contrib.messages.middleware.MessageMiddleware',
+#     'django.middleware.clickjacking.XFrameOptionsMiddleware',
+# ]
+
+MIDDLEWARE = [
     'corsheaders.middleware.CorsMiddleware',
+    'django.middleware.security.SecurityMiddleware',
```
Comment (crwallace) [inline config/settings/common.py:62]: Does `djangosecure.middleware.SecurityMiddleware` no longer need to be listed first?

File: config/settings/production.py [hunk 9]
```diff
@@ -40,10 +40,11 @@
 # Hosts/domain names that are valid for this site
 # See https://docs.djangoproject.com/en/1.6/ref/settings/#allowed-hosts
 ALLOWED_HOSTS = env.list('DJANGO_ALLOWED_HOSTS', default=['*'])
+# ALLOWED_HOSTS = ['0.0.0.0']
```
Comment (crwallace) [inline config/settings/production.py]: Extra comment

File: apps/jurisdiction/migrations/0001_initial.py (hunk 1 above)
Comment (MagnetonIO) [reply apps/jurisdiction/migrations/0001_initial.py:57]: Required argument since Django 2

File: config/urls.py [hunk 10]
```diff
@@ -15,14 +15,15 @@
...
 
 
-urlpatterns = patterns(
-    '',
+urlpatterns = [
     # Uncomment the next line to enable the admin:
     url(r'^jet/', include('jet.urls', 'jet')),
     url(r'^jet/dashboard/', include('jet.dashboard.urls', 'jet-dashboard')),  # Django JET dashboard URLS
-    url(r'^admin/', include(admin.site.urls)),
+    url(r'^admin/', admin.site.urls),
```
Comment (crwallace) [inline config/urls.py:22]: Why was `include()` removed from the admin?

File: docker-compose.yml [hunk 11]
```diff
@@ -25,7 +25,9 @@ services:
     <<: *base_django_setup
     ports:
       - 8000:8000
-    command: python manage.py runserver 0.0.0.0:8000
+    command: python manage.py runserver 0.0.0.0:8000 --noreload
```
Comment (crwallace) [inline docker-compose.yml:28]: What does the `--noreload` flag mean?

**Review by crwallace — COMMENTED** [review]
Thank you so much! Overall this looks great! I have a few questions and a few nitpicks...

File: apps/jurisdiction/migrations/0001_initial.py (hunk 1 above)
Comment (crwallace) [reply apps/jurisdiction/migrations/0001_initial.py:57]: Gotcha!

Note to my future self here: https://code.djangoproject.com/ticket/28677

File: config/settings/common.py (hunk 8 above)
Comment (MagnetonIO) [reply config/settings/common.py:62]: the order doesn't matter

File: docker-compose.yml (hunk 11 above)
Comment (MagnetonIO) [reply docker-compose.yml:28]: Stop autoreload if you make changes in dev

File: config/urls.py (hunk 10 above)
Comment (MagnetonIO) [reply config/urls.py:22]: In Django 1.9+, you don't need to use include

Comment (crwallace) [reply config/urls.py:22]: Gotcha! Thanks

**Review by crwallace — APPROVED** [review]
This looks great!

**Pull request merged**